### PR TITLE
[PPP-4865] - Snakeyaml and Karaf Update - Karaf 4.2.15

### PR DESCRIPTION
**Pull request by andreramos89**
@smmribeiro 
@renato-s 

Note this PR includes not only safeyaml but also jackson-dataformat-yaml upgrade

### CODE REVIEW

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 9m 30s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -Djs.no.sandbox -pl \
assemblies/common-resources,assemblies/server
```
---
 :exclamation: No tests found!
 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 4m 14s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -Djs.no.sandbox -pl \
assemblies/common-resources,assemblies/server
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

File: assemblies/server/src/main/resources-filtered/etc/custom.properties [hunk 1]
```diff
@@ -137,7 +137,7 @@ org.osgi.framework.system.packages.extra= \
  org.slf4j.*; version="${slf4j.version}", \
  org.springframework.dao; version="${spring.version}", \
  org.springframework.security.*; version="${spring-security.version}", \
- org.yaml.snakeyaml; version="${snakeyaml.version}", \
+ org.yaml.snakeyaml; version="${safeyaml.version}", \
```
Comment (smmribeiro) [inline assemblies/server/src/main/resources-filtered/etc/custom.properties:140]: Shouldn't the groupId be "org.hitachivantara"?

File: assemblies/common-resources/src/main/resources-filtered/etc/custom.properties [hunk 2]
```diff
@@ -151,7 +151,7 @@ org.osgi.framework.system.packages.extra= \
  org.slf4j.*; version="${slf4j.version}", \
  org.springframework.dao; version="${spring.version}", \
  org.springframework.security.*; version="${spring-security.version}", \
- org.yaml.snakeyaml; version="${snakeyaml.version}", \
+ org.yaml.snakeyaml; version="${safeyaml.version}", \
```
Comment (smmribeiro) [inline assemblies/common-resources/src/main/resources-filtered/etc/custom.properties:154]: Shouldn't the groupId be "org.hitachivantara"?

**Review by smmribeiro — CHANGES_REQUESTED** [review]

**Review by smmribeiro — APPROVED** [review]
Looks good.

File: assemblies/server/src/main/resources-filtered/etc/custom.properties (hunk 1 above)
Comment (andreramos89) [reply assemblies/server/src/main/resources-filtered/etc/custom.properties:140]: Not really.

File: assemblies/common-resources/src/main/resources-filtered/etc/custom.properties (hunk 2 above)
Comment (andreramos89) [reply assemblies/common-resources/src/main/resources-filtered/etc/custom.properties:154]: Not really

**Pull request merged**